src/main/java/frc/robot/subsystems/gripper/Gripper.java

```diff
@@ -0,0 +1,73 @@
+package frc.robot.subsystems.gripper;
+
+import edu.wpi.first.networktables.BooleanSubscriber;
+import edu.wpi.first.networktables.DoubleSubscriber;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Ports;
+import frc.robot.subsystems.LoggedSubsystem;
+
+public class Gripper extends LoggedSubsystem<GripperLoggedInputs> {
+    private GripperLoggedInputs gripperLoggedInputs;
+    private final Solenoid leftSolenoid = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Gripper.LEFT_SOLENOID);
+    private final Solenoid rightSolenoid = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Gripper.RIGHT_SOLENOID);
+
+    private final NetworkTable table = NetworkTableInstance.getDefault().getTable("gripper");
+
+
+    public Gripper(GripperLoggedInputs inputs) {
+        super(inputs);
+    }
+
+    /**
+     * setSolenoid sets the Solenoid state
+     * @param state
+     */
+
+    public void setSolenoid(boolean state) {
+        leftSolenoid.set(state);
+        rightSolenoid.set(state);
+
+    }
+
+    /**
+     * changes the current state to the other state
+     */
+
+    public void toggleSolenoid() {
+        leftSolenoid.toggle();
+        rightSolenoid.toggle();
+
+    }
+
+    /**
+     * return the current state of the solenoids
+     */
+    public void getRightSolenoid() {
+        rightSolenoid.get();
+    }
+
+    public void getLeftSolenoid() {
+        leftSolenoid.get();
+    }
```

Comment:
What is a use case for these methods?

Reply:
Well, the only case I can think of is when the arm engages in order to pick up cones, if we want to do that we should know whether the gripper is closed or open.

Reply:
You get a parameter `GripperLogInputs` in the constructor, and pass it to the super constructor. However, the logger inputs should be instantiated right in the constructor, without being passed as a parameter.